auth/client-provider/src/main/java/org/wildfly/security/auth/client/provider/ElytronClientDefaultSSLContextProvider.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2021 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.auth.client.provider;
+
+import org.kohsuke.MetaInfServices;
+import org.wildfly.client.config.ConfigXMLParseException;
+import org.wildfly.security.auth.client.AuthenticationContext;
+import org.wildfly.security.auth.client._private.ElytronMessages;
+import static org.wildfly.security.auth.client._private.ElytronMessages.log;
+
+import java.io.FileNotFoundException;
+import java.security.GeneralSecurityException;
+import java.security.NoSuchAlgorithmException;
+import java.security.Provider;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Provider that loads Elytron client configuration and provides default SSLContext which can be returned with SSLContext.getDefault() call.
+ * Default SSLContext is the configured SSL context that does not have any specific rule when it should be used, so it matches all rules.
+ */
+@MetaInfServices(value = Provider.class)
+public final class ElytronClientDefaultSSLContextProvider extends Provider {
+
+    public static final String ELYTRON_CLIENT_DEFAULT_SSL_CONTEXT_PROVIDER_NAME = "ElytronClientDefaultSSLContextProvider";
+
+    public ElytronClientDefaultSSLContextProvider() {
```

Comment:
Would be good to add javadoc for the constructors.

Reply:
Addd